src/globus_sdk/services/gcs/connector_table.py

```diff
@@ -0,0 +1,147 @@
+from __future__ import annotations
+
+import dataclasses
+import re
+import typing as t
+
+from globus_sdk._types import UUIDLike
+
+_NORMALIZATION_PATTERN = re.compile(r"[_\- ]+")
+
+
+def _normalize_name(name: str) -> str:
+    return _NORMALIZATION_PATTERN.sub("-", name.strip()).lower()
```

Comment:
Admittedly ends up being a very "permissive" normalization (afa resulting in matching on some crazy strings) but I also don't see any reason to expect this will be harmful. Given that users might build apps that looks these up by crazy names (and thus rely on this normalization) are we comfortable committing to this relatively long term (in software lifecycle time)?

Reply:
I am, in that the rule can be explained pretty simply as
- `_`, `-`, and spaces are all equivalent
- runs of those things of any non-zero length are equivalent

So I don't think we're tied to this particular implementation -- which is the main thing which would give me discomfort.